fontforge/splinefit.c

```diff
@@ -731,248 +731,315 @@ return( SplineMake3(from,to));
     }
+    /* This is the generic case, where a generic part is approximated by a cubic */
+    /* bezier spline. */
+	if ( is_accurate ) { /* More accurate but slower function by Linus Romer */
+		bigreal best_error = 1e30;
+		bigreal t,error,errorsum,dist;
+		BasePoint prevcp,coeff1,coeff2,coeff3;
+		bigreal best_fromhandle = 0.0;
+		bigreal best_tohandle = 0.0;
+		BasePoint approx[99]; /* The 99 points on the approximate cubic bezier */
+		/* We make 2 runs: The first run to narrow the variation range, the second run to finetune */
+		for (int run=0; run<2; ++run) { 
+			for (int fromhandle=((run==0)?1:-30); fromhandle<=((run==0)?60:30); ++fromhandle) {
+				for (int tohandle=((run==0)?1:-30); tohandle<=((run==0)?60:30); ++tohandle) {
+					nextcp.x = from->me.x+ftlen*fromunit.x*( (run==0)?fromhandle:best_fromhandle+fromhandle/30.0 )/60.0;
+					nextcp.y = from->me.y+ftlen*fromunit.y*( (run==0)?fromhandle:best_fromhandle+fromhandle/30.0 )/60.0;
+					prevcp.x = to->me.x+ftlen*tounit.x*( (run==0)?tohandle:best_tohandle+tohandle/30.0 )/60.0;
+					prevcp.y = to->me.y+ftlen*tounit.y*( (run==0)?tohandle:best_tohandle+tohandle/30.0 )/60.0;
+					/* Calculate the error of the cubic bezier path from,nextcp,prevcp,to: */
+					/* In order to do that we calculate 99 points on the bezier path. */
+					coeff3.x = -from->me.x+3*nextcp.x-3*prevcp.x+to->me.x;
+					coeff3.y = -from->me.y+3*nextcp.y-3*prevcp.y+to->me.y;
+					coeff2.x = 3*from->me.x-6*nextcp.x+3*prevcp.x;
+					coeff2.y = 3*from->me.y-6*nextcp.y+3*prevcp.y;
+					coeff1.x = -3*from->me.x+3*nextcp.x;
+					coeff1.y = -3*from->me.y+3*nextcp.y;
+					for (int i=0; i<99; ++i) {
+						t = (i+1)/100.0;
+						approx[i].x = from->me.x+t*(coeff1.x+t*(coeff2.x+t*coeff3.x));
+						approx[i].y = from->me.y+t*(coeff1.y+t*(coeff2.y+t*coeff3.y));
+					}
+					/* Now we calculate the error by determing the minimal quadratic distance to the mid points. */
+					errorsum = 0.0;
+					for (int i=0; i<cnt; ++i) { /* Going through the mid points */
+						error = (mid[i].p.x-approx[0].x)*(mid[i].p.x-approx[0].x)
+						+(mid[i].p.y-approx[0].y)*(mid[i].p.y-approx[0].y);
+						/* Above we have just initialized the error and */
+						/* now we are going through the remaining 98 of */
+						/* 99 points on the approximate cubic bezier: */
+						for (int j=1; j<99; ++j) { 
+							dist = (mid[i].p.x-approx[j].x)*(mid[i].p.x-approx[j].x)
+							+(mid[i].p.y-approx[j].y)*(mid[i].p.y-approx[j].y);
+							if (dist < error)
+								error = dist;
+						}
+						errorsum += error;
+						if (errorsum > best_error)
+							break;
+					}
+					if (errorsum < best_error) {
+						best_error = errorsum;
+						if (run == 0) {
+							best_fromhandle = fromhandle;
+							best_tohandle = tohandle;
+						}
+						from->nextcp = nextcp;
+						to->prevcp = prevcp;
+					}
+				}
+			}
+		}
+return( SplineMake3(from,to));	
+	}
+	else { /* original and fast function */
+		pt_pf_x = to->me.x - from->me.x;
```

Comment:
In the past some project members have supported not changing the indentation of large blocks in cases like this so as not to change `git blame` ownership (among other reasons). I'm agnostic about shorter blocks but I think for something this large it would be better not to change the lines. 

The lines around the block can be left as-is (as far as I am concerned).